### PR TITLE
Update next.js bundle.yaml with env and metadata sections

### PR DESCRIPTION
**Pull request by sjjj986**
- pass user's nextjs version from framework buildpack
- add an empty env section
- add a metadata section with adapter & nextjs metrics
- refactor bundle.yaml test

### CODE REVIEW

File: packages/@apphosting/adapter-nextjs/src/bin/build.ts [hunk 1]
```diff
@@ -15,11 +15,21 @@ const opts = getBuildOptions();
 process.env.NEXT_PRIVATE_STANDALONE = "true";
 // Opt-out sending telemetry to Vercel
 process.env.NEXT_TELEMETRY_DISABLED = "1";
+// Assert framework version exists
```
Comment (tonyjhuang) [inline packages/@apphosting/adapter-nextjs/src/bin/build.ts:18]: small nit: imo this comment is unnecessary, the code is straightforward enough

File: packages/@apphosting/adapter-nextjs/src/interfaces.ts [hunk 2]
```diff
@@ -101,3 +101,11 @@ export interface OutputBundleOptions {
    */
   outputStaticDirectoryPath: string;
 }
+
+// Metadata schema for bundle.yaml outputted by next.js adapter
+export interface Metadata {
+  adapterNpmPackageName: string;
```
Comment (tonyjhuang) [inline packages/@apphosting/adapter-nextjs/src/interfaces.ts:107]: ooc how does this change when we support non node dependencies, eg for python apps?

File: packages/@apphosting/adapter-nextjs/src/utils.ts [hunk 3]
```diff
@@ -112,11 +113,30 @@ async function moveResources(appDir: string, outputBundleAppDir: string): Promis
   return;
 }
 
+/**
+ * Create metadata needed for outputting adapter and framework metrics in bundle.yaml.
+ */
+export function createMetadata(nextVersion: string): Metadata {
+  const __dirname = dirname(fileURLToPath(import.meta.url));
```
Comment (tonyjhuang) [inline packages/@apphosting/adapter-nextjs/src/utils.ts:120]: why __dirname and not dirname?

**Review by tonyjhuang — CHANGES_REQUESTED** [review]

File: packages/@apphosting/adapter-nextjs/src/interfaces.ts (hunk 2 above)
Comment (sjjj986) [reply packages/@apphosting/adapter-nextjs/src/interfaces.ts:107]: We can modify the output bundle spec naming when needed? Revisiting in discussion meeting.

Comment (sjjj986) [reply packages/@apphosting/adapter-nextjs/src/interfaces.ts]: Discussed and agreed to make this field more generic, changing to`adapterPackageName`

**Review by tonyjhuang — APPROVED** [review]

**Pull request merged**